### PR TITLE
Correctly preserve existing vite envPrefix config

### PR DESCRIPTION
**Pull request by bodograumann**
## What I did

I changed the way the `envPrefix` config option in the vite builder is set, as discussed on discord, @IanVS ;-)
Now that it is an array, `mergeConfig` from vite will always prepend any existing `envPrefix` options from the users vite config.

## How to test

I included a unit test which only succeeds with the changes I made.
The dummy options I used there are found mostly through trial and error, because I couldn't understand all the magic and types in `core-commons/preset`. So I am open for any suggestions.

## Checklist

- [x] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [ ] Make sure to add/update documentation regarding your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] If this PR should be tested against many or all sandboxes,
      make sure to add the `ci:merged` or `ci:daily` GH label to it.
- [ ] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

**Comment by bodograumann** [comment]
I have no idea why the tests fail here, when they work locally :-/

**Comment by IanVS** [comment]
I'm seeing the same.  I set my node to 16.19.0, pulled down the branch, installed, compiled (with `--no-link` just like in CI), and running the same command as ci (`yarn test --runInBand --ci`) passes just fine.  @joshwooding when you were working on the PR for MDX options, did you run across anything like this?  It's a shot in the dark, but I'm not sure what we can do to fix the failing test.  😕

File: code/lib/builder-vite/src/vite-config.test.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  presets: {
+    apply: async (key: string) =>
+      ({
+        framework: {
+          name: '',
+        },
+        addons: [],
+        core: {
+          builder: {},
+        },
```
Comment (IanVS) [inline code/lib/builder-vite/src/vite-config.test.ts:24]: What if you add `options: {}` here?  Maybe that would fix the test?

Comment (bodograumann) [reply code/lib/builder-vite/src/vite-config.test.ts:24]: Yes that works. Thanks for the idea! :-)

**Review by IanVS — APPROVED** [review]
This is awesome, I love the unit tests, thanks so much for doing that.

**Review by ndelangen — APPROVED** [review]

**Pull request merged**